cosmology.c

```diff
@@ -57,15 +51,12 @@ double hubble_function(double a)
     
     hubble_a += OmegaFLD(a);
     hubble_a += CP->OmegaK / (a * a);
-    hubble_a += CP->Omega0 / (a * a * a);
+    hubble_a += (CP->OmegaCDM + CP->OmegaBaryon) / (a * a * a);
 
     if(CP->RadiationOn) {
         hubble_a += CP->OmegaG / (a * a * a * a);
-        /* massless neutrinos are added only if there is no (massive) neutrino particle.*/
-        if(CP->MasslessNeutrinosOn)
-            hubble_a += CP->OmegaNu0 / (a * a * a * a);
+        hubble_a += get_omega_nu(&CP->ONu, a);
```

Comment:
What about the massive neutrino?

Reply:
It currently is not in OmegaCDM.

Reply:
if get_omega_nu returns the sum of massive and massless nu, this line looks even stranger ... aren't you supposed to divide this by `a**3` or `a**4`?

Reply:
No, it returns the sum of the massive neutrinos as a redshift dependent property: OmegaNu(a). For high masses we have OmegaNu(a) ~ MNu/(93.15 h^2) and for low masses OmegaNu(a) ~a^-4. Because the neutrino mass interpolates between the two (noticeably so) it seemed the least confusing thing to do

Reply:
This touches how k-space-neutrino is implemented and I don't think it can be addressed in this PR; especially given that you are also embedding it in other places..

If you think of neutrino as a single fluid then yes, I see how the current get_omega_nu is the least confusing approach. massive neutrino still has a pressure component that scales like a^-4. I think that's likely what you mean by the neutrino mass interpolates between the two. 

However, after studying class somewhat carefully, I tend to agree with class where it divides neutrinos into components in terms of how they scale with a: ur (a**-4), mass-like (a**-3), pressure-like (a**-4), like here:

https://github.com/lesgourg/class_public/blob/master/source/background.c#L354

Reply:
I think the reason I like the single fluid is that it matches how I am treating the perturbations, and it matches well with how particles are treated in neutrino particle simulations: they are a single fluid, described by particles. It also has a neat interpretation in terms of the maximum neutrino velocity for the hybrid particles: you can set Max_nuvel = C/4 (say) and then the relativistic component is taken care of automatically.

Reply:
I see. The hybrid particles deals with mass-like and pressure-like massive neutrinos the same time. What about the ultra relativistic species?

Reply:
Not sure what you mean exactly? pressure-like massive neutrinos are the ultra-relativistic species?

Reply:
No. I mean there is a pressure-like component to the overall density due to massive neutrino that also scales like a**-4, in this line:
https://github.com/lesgourg/class_public/blob/master/source/background.c#L380

ultra-relativstic is already a**-4, at:
https://github.com/lesgourg/class_public/blob/master/source/background.c#L418

Reply:
Ah, I see. So you are proposing adding support for an arbitrary amount of ultra-relativistic species? 

At present we have OmegaG, which is so called because it is actually photons (and the parameter we set is the CMBTemperature). I think just renaming it OmegaUR would be not the right thing to do, because it is set up to be actual photons. But I think you are suggesting adding an extra species that redshifts like a^-4 and could map onto, for example, dark photons?